scripts/west_commands/build.py

```diff
@@ -231,7 +234,7 @@ def _setup_build_dir(self):
 
         self.build_dir = build_dir
 
-    def _setup_source_dir(self):
+    def _find_source_dir(self):
         # Initialize source_dir attribute, either from command line argument,
```

Comment:
This comment is no longer true after this change.

In general looking at this file with fresh eyes, there are places where we're using self.args.source_dir where we should be using self.source_dir (e.g. in _parse_remainder). The former is what the user said, if anything, the latter is what we decided the answer is.

The purpose of this function is to set self.source_dir, but it seems to be being extended to be used differently. The purpose is to find the source directory so it can be used in the source_dir format string when creating the build directory.

This is starting to get a bit messy and hard to follow. I think it would be better combined into a single `_setup_dirs()` function that sets `self.source_dir`, `self.build_dir`, and `self.created_build_dir`. That will clean up the code around the "legacy -s  option" comment as well. This routine should probably also initialize the CMake cache if the build directory exists.

Reply:
>In general looking at this file with fresh eyes, there are places where we're using self.args.source_dir where we should be using self.source_dir (e.g. in _parse_remainder). The former is what the user said, if anything, the latter is what we decided the answer is.

I don't think so, in `_parse_remainder()` we're just using `self.args.source_dir` as a temporary variable. That exists because we wanted to keep compatibility with `-s`.
Regarding the cleanup, I agree with you, but I'd rather not do it in this PR. I don't think the changes in this PR worsen the current state of things in any significant way, and I have more changes in mind for the near future that will require the cleanup.

Reply:
> I have more changes in mind for the near future that will require the cleanup.

If that means you're willing do to the cleanup I'm OK with it. The current change has crossed my personal OCD threshold ;).